theano/tensor/opt.py

```diff
@@ -3346,6 +3475,12 @@ def local_join_empty(node):
             assert ret.dtype == o.dtype
             assert ret.ndim == o.ndim
             ret = T.patternbroadcast(ret, node.outputs[0].broadcastable)
+
+        # Copy over stacktrace from previous output (after join op)
+        # to new output, because an error in the new op must be caused
+        # by an error in the old join op.
+        copy_stack_trace(node.outputs, ret)
```

Comment:
Here, I would add the stacktrace to `ret` before checking the type (when it is the output of the new `T.Join`, and then add it also to the output of `T.patternbroadcast`. Otherwise, if the new Join fails but there is a patternbroadcast afterwards, there will be no trace on the Join.